Mathlib/NumberTheory/LSeries/DirichletContinuation.lean

```diff
@@ -103,6 +104,87 @@ lemma Even.LFunction_neg_two_mul_nat {χ : DirichletCharacter ℂ N} (hχ : Even
     LFunction χ (-(2 * n) - 1) = 0 :=
   ZMod.LFunction_neg_two_mul_nat_sub_one hχ.to_fun n
 
+/-!
+## Results on changing levels
+-/
+
+private lemma LFunction_changeLevel_aux {M N : ℕ} [NeZero M] [NeZero N] (hMN : M ∣ N)
+    (χ : DirichletCharacter ℂ M) {s : ℂ} (hs : s ≠ 1) :
+    LFunction (changeLevel hMN χ) s =
+      LFunction χ s * ∏ p ∈ N.primeFactors, (1 - χ p * p ^ (-s)) := by
+  have hpc : IsPreconnected ({1}ᶜ : Set ℂ) :=
+    (isConnected_compl_singleton_of_one_lt_rank (rank_real_complex ▸ Nat.one_lt_ofNat) _)
+      |>.isPreconnected
+  have hne : 2 ∈ ({1}ᶜ : Set ℂ) := by norm_num
+  refine AnalyticOnNhd.eqOn_of_preconnected_of_eventuallyEq (𝕜 := ℂ)
+    (g := fun s ↦ LFunction χ s * ∏ p ∈ N.primeFactors, (1 - χ p * p ^ (-s))) ?_ ?_ hpc hne ?_ hs
+  · refine DifferentiableOn.analyticOnNhd (fun s hs ↦ ?_) isOpen_compl_singleton
+    exact (differentiableAt_LFunction _ _ (.inl hs)).differentiableWithinAt
+  · refine DifferentiableOn.analyticOnNhd (fun s hs ↦ ?_) isOpen_compl_singleton
+    refine ((differentiableAt_LFunction _ _ (.inl hs)).mul ?_).differentiableWithinAt
+    refine .finset_prod fun i h ↦ (differentiableAt_const _).sub ((differentiableAt_const _).mul ?_)
+    have : NeZero i := ⟨(Nat.pos_of_mem_primeFactors h).ne'⟩
+    apply differentiable_const_cpow_neg
+  · refine eventually_of_mem ?_  (fun t (ht : 1 < t.re) ↦ ?_)
+    · exact (continuous_re.isOpen_preimage _ isOpen_Ioi).mem_nhds (by norm_num : 1 < (2 : ℂ).re)
+    · simpa only [LFunction_eq_LSeries _ ht] using LSeries_changeLevel hMN χ ht
+
+/-- If `χ` is a Dirichlet character and its level `M` divides `N`, then we obtain the L function
+of `χ` considered as a Dirichlet character of level `N` from the L function of `χ` by multiplying
+with `∏ p ∈ N.primeFactors, (1 - χ p * p ^ (-s))`.
+(Note that `1 - χ p * p ^ (-s) = 1` when `p` divides `M`). -/
+lemma LFunction_changeLevel {M N : ℕ} [NeZero M] [NeZero N] (hMN : M ∣ N)
+    (χ : DirichletCharacter ℂ M) {s : ℂ} (h : χ ≠ 1 ∨ s ≠ 1) :
+    LFunction (changeLevel hMN χ) s =
+      LFunction χ s * ∏ p ∈ N.primeFactors, (1 - χ p * p ^ (-s)) := by
+  rcases h with h | h
+  · have hχ : changeLevel hMN χ ≠ 1 := h ∘ (changeLevel_eq_one_iff hMN).mp
+    have h' : Continuous fun s ↦ LFunction χ s * ∏ p ∈ N.primeFactors, (1 - χ p * ↑p ^ (-s)) :=
+      (differentiable_LFunction h).continuous.mul <|
+        continuous_finset_prod _ fun p hp ↦ continuous_const.sub <| continuous_const.mul <|
+          have : NeZero p := ⟨(Nat.prime_of_mem_primeFactors hp).ne_zero⟩;
+          continuous_const_cpow_neg _
+    exact congrFun ((differentiable_LFunction hχ).continuous.ext_on
+      (dense_compl_singleton 1) h' (fun _ h ↦ LFunction_changeLevel_aux hMN χ h)) s
+  · exact LFunction_changeLevel_aux hMN χ h
+
+/-!
+## The `L`-function of the trivial character mod `N`
+-/
+
+/-- The `L`-function of the trivial character mod `N`. -/
+noncomputable abbrev LFunction_triv_char (N : ℕ) [NeZero N] :=
```

Comment:
You changed `LFunction_one` to `LFunction_triv_char` here, but not in the lemma names below.
Since it is define as `LFunction 1`, I think `LFunction_one` may actually be the better name. In any case, lemma names referring to it should be consistent with its name.

Reply:
Was the decision to stick with `LFunction_triv_char` recorded somewhere?

Anyways, since this is a def (abbrev) I think it should be CamelCased.
```suggestion
noncomputable abbrev LFunctionTrivChar (N : ℕ) [NeZero N] :=
```
or possibly
```suggestion
noncomputable abbrev LFunctionOne (N : ℕ) [NeZero N] :=
```

Reply:
Michael's approval tick and message "looks good to me now" was after this, so I took that to include approval of my decision to stick with "triv char" and adjust the lemma names to match. (I don't like "LFunction_one" since it depends on exactly _which_ 1 you are talking about – it is the "1" of the group of Dirichlet characters mod N, and the value of N definitely matters.)